### DISCUSSION TITLE
Developer Getting Started Guide yields: "There was an uncaught error: The provided API key is invalid."

### DISCUSSION THREAD

**Discussion by tony-bpc** [danny-avila/LibreChat#3156] [Troubleshooting]
### What happened?

I followed this guide and am blocked by errors:
 
https://www.librechat.ai/docs/development/get_started
 
I tried historical versions resulting in same issue (0.7.2 , 0.6.5)
 
I created a Mongo DB connection, swapped default credentials, and created clean codespaces. These approaches did not solve the error.
 
Are others having this issue?

### Steps to Reproduce

I used the following commands:

cd LibreChat/
npm ci
npm run frontend 
npm run backend
----> error

Also tried:
npm run reinstall
npm run backend
-----> same error

### What browsers are you seeing the problem on?

Microsoft Edge

### Relevant log output

```shell
vscode@713d1ee7fdea:/workspaces/LibreChat$ npm run backend

> LibreChat@0.6.5 backend
> cross-env NODE_ENV=production node api/server/index.js

2024-06-21 21:26:45 info: `REDIS_URI` not provided, or `USE_REDIS` not set. Redis module will not be initialized.
2024-06-21 21:26:46 error: There was an uncaught error: The provided API key is invalid.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by berry-13** [comment]
skipping this error, does the server remain active? It's hard to understand, but it doesn't seem to be a MongoDB authentication error. can you try to run LibreChat with the default `.env` except for the Mongo URI?

**Comment by danny-avila** [comment]
That's weird, I use the setup as described in the guide and have never encountered this.

Can you try with the default env setup as shown in `.env.example` file as @berry-13 mentioned?

**Reply by tony-bpc** [reply]
This result was based on a default .env setup. Resulting in the invalid API key error.

**Comment by fuegovic** [comment]
weird, it's also working fine for me 

- `git clone https://github.com/danny-avila/LibreChat.git`
- `cp .env.example .env`
  - fill in my mongodb atlas URI
- `npm ci`
- `npm run frontend`
- `npm run backend`

```sh
PS D:\v0.7.4rc1\LibreChat> npm run backend

> LibreChat@0.7.4-rc1 backend
> cross-env NODE_ENV=production node api/server/index.js

2024-06-21 17:53:44 info: [Optional] Redis not initialized. Note: Redis support is experimental.
2024-06-21 17:53:46 warn: Meilisearch error, search will be disabled
2024-06-21 17:53:47 info: Connected to MongoDB
2024-06-21 17:53:47 error: [indexSync] error fetch failed
2024-06-21 17:53:49 error: Config file YAML format is invalid: ENOENT: no such file or directory, open 'D:\v0.7.4rc1\LibreChat\librechat.yaml'
2024-06-21 17:53:49 warn: Default value for CREDS_KEY is being used.
2024-06-21 17:53:49 warn: Default value for CREDS_IV is being used.
2024-06-21 17:53:49 warn: Default value for JWT_SECRET is being used.
2024-06-21 17:53:49 warn: Default value for JWT_REFRESH_SECRET is being used.
2024-06-21 17:53:49 info: Please replace any default secret values.
2024-06-21 17:53:49 info: 

    For your convenience, use this tool to generate your own secret values:
    https://www.librechat.ai/toolkit/creds_generator

    ​
2024-06-21 17:53:49 warn: RAG API is either not running or not reachable at undefined, you may experience errors with file uploads.
2024-06-21 17:53:49 info: Server listening at http://localhost:3080
```

**Comment by berry-13** [comment]
@tony-bpc seems to be an error with codespace, not sure how to solve this

![image](https://github.com/danny-avila/LibreChat/assets/81851188/38c190dc-5522-4f7d-9a5a-e700e6f5f603)

**Reply by tony-bpc** [reply]
This error occurs still in VS Code same process as @fuegovic above.
```
PS C:\Users\tpappas\Downloads\LibreChat-main\LibreChat-main> npm run backend

> LibreChat@0.7.4-rc1 backend
> cross-env NODE_ENV=production node api/server/index.js

node:buffer:319
  throw new ERR_INVALID_ARG_TYPE(
  ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (node:buffer:319:9)
    at Object.<anonymous> (C:\Users\tpappas\Downloads\LibreChat-main\LibreChat-main\api\server\utils\crypto.js:4:20)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (C:\Users\tpappas\Downloads\LibreChat-main\LibreChat-main\api\server\utils\index.js:7:21)
    at Module._compile (node:internal/modules/cjs/loader:1358:14) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v20.14.0
```

**Reply by berry-13** [reply]
no, here you didn't create the .env or the librechat.yaml

**Reply by tony-bpc** [reply]
I had created the .env file and tried this line cp .env.example .env which got LibreChat to run locally now through VS code. Still have the same issue in codespaces. Thank you for your help.

**Comment by danny-avila** [comment]
"The provided API key is invalid." isn't anywhere in the code, so it's from an external connection? Upon searching on google, it seems to be a common "google" API message. 

Try commenting every variable in the .env that starts with `GOOGLE_` just as a good measure.

There are no API connections upon startup aside from mongodb that would be initialized, so maybe codespaces has some google analytics built into the environment?

**Reply by basnijholt** [reply]
For me this turned out to be the issue.

**Reply by basnijholt** [reply]
Actually, it seemed to be a coincidence that removing the GOOGLE API key fixed the issue. I am still running into this.

EDIT: I found the solution, see https://github.com/danny-avila/LibreChat/discussions/3156#discussioncomment-12430838

**Comment by raymelon** [comment]
I also encountered the same issue when cloning the LibreChat space in HuggingFace

I followed this guide: https://www.librechat.ai/docs/remote/huggingface

**Comment by ziyuyouming** [comment]
You can try modifying the `MEILI_MASTER_KEY` in the `.devcontainer/docker-compose.yml` file. This configuration item also exists in `.env.example`. You can get the value of `MEILI_MASTER_KEY` in `.env.example` and then update it in the `.devcontainer/docker-compose.yml` file.

` .devcontainer/docker-compose.yml`:

```yaml
meilisearch:
    ...
    environment:
      ...
-     - MEILI_MASTER_KEY=5c71cf56d672d009e36070b5bc5e47b743535ae55c818ae3b735bb6ebfb4ba63
+     - MEILI_MASTER_KEY=DrhYf7zENyR6AlUCKmnz0eYASOQdl6zxH7s7MKFSfFCt
```

**Reply by owengo** [reply]
I had also the error and it was because the MEILI_MASTER_KEY was incorrect.

**Comment by basnijholt** [comment]
**SOLUTION FOUND**

tl;dr: you have another docker service called `meilisearch`!

I keep running into this issue. Sometimes it works but then after restarting I get this error.

I figured out that commenting out `MEILI_MASTER_KEY=DrhYf7zENyR6AlUCKmnz0eYASOQdl6zxH7s7MKFSfFCt` will allow the container to start, however, search will work. I then entered the `chat-meilisearch` container and checked out `env` and verified that the API key is correct.

That led me to the following realization, I host another container with the same name `meilisearch`, and since it is on the same docker network, it tries to connect to that one.

Simply changing this in my `compose.yaml`:
```diff
diff --git a/librechat/compose.yaml b/librechat/compose.yaml
index 63ab5eb..f0a4baa 100644
--- a/librechat/compose.yaml
+++ b/librechat/compose.yaml
@@ -17,7 +17,7 @@ services:
     environment:
       - HOST=0.0.0.0
       - MONGO_URI=mongodb://mongodb:27017/LibreChat
-      - MEILI_HOST=http://meilisearch:7700
+      - MEILI_HOST=http://meilisearchchat:7700
       - RAG_PORT=${RAG_PORT:-8000}
       - RAG_API_URL=http://rag_api:${RAG_PORT:-8000}
     volumes:
@@ -51,7 +51,7 @@ services:
       - /mnt/data/librechat/data-node:/data/db
     command: mongod --noauth

-  meilisearch:
+  meilisearchchat:
     container_name: chat-meilisearch
     networks:
       - mynetwork
@@ -59,7 +59,7 @@ services:
     restart: always
     user: "${UID}:${GID}"
     environment:
-      - MEILI_HOST=http://meilisearch:7700
+      - MEILI_HOST=http://meilisearchchat:7700
       - MEILI_NO_ANALYTICS=true
       - MEILI_MASTER_KEY=${MEILI_MASTER_KEY}
     volumes:
```

Fixed the issue :tada:
